src/Polly.Core/CircuitBreaker/BrokenCircuitException.cs

```diff
@@ -47,7 +47,7 @@ public BrokenCircuitException(string message, Exception inner)
     /// </summary>
     /// <param name="info">The <see cref="SerializationInfo" /> that holds the serialized object data about the exception being thrown.</param>
     /// <param name="context">The <see cref="StreamingContext" /> that contains contextual information about the source or destination.</param>
-    protected BrokenCircuitException(SerializationInfo info, StreamingContext context)
+    protected private BrokenCircuitException(SerializationInfo info, StreamingContext context)
```

Comment:
The class isn't sealed though, so someone might have derived from it and implemented the same pattern.

Reply:
The problem I had here is that this API doesn't exists on .NET Core and it would force us to duplicate the `PublicAPI.Shipped` and `PublicAPI.Unshipped` fiels for each target framework which I don't think is worth it.

I can just revert the changes and suppress it on this API. 

Alternatively, we can drop the binary support? It was on the plan anyway, or not?

Reply:
I thought we were leaving it in for v8 for .NET Framework compatibility.

Reply:
Let me just put the constructor back (just in case).